build.sbt

```diff
@@ -8,25 +8,27 @@ crossPaths := false
 
 publishMavenStyle := true
 
-publishTo <<= version { (v: String) =>
+publishTo := {
   val nexus = "https://oss.sonatype.org/"
-  if (v.trim.endsWith("SNAPSHOT")) Some("snapshots" at nexus + "content/repositories/snapshots")
-  else Some("releases" at nexus + "service/local/staging/deploy/maven2")
+  if (isSnapshot.value)
+    Some("snapshots" at nexus + "content/repositories/snapshots")
+  else
+    Some("releases"  at nexus + "service/local/staging/deploy/maven2")
 }
 
-scalaVersion := "2.11.2"
+scalaVersion := "2.12.2"
 
 scalacOptions ++= Seq( "-deprecation", "-unchecked", "-feature", "-language:postfixOps", "-language:existentials", "-language:implicitConversions"  )
 
-javacOptions ++= Seq("-source", "1.7")
+javacOptions ++= Seq("-source", "1.8")
 
 parallelExecution in Test := false
 
 libraryDependencies ++= Seq(
-  "junit" % "junit" % "4.11" % "provided, test",
-  "com.google.guava" % "guava" % "14.0" % "provided",
+  "junit" % "junit" % "4.12" % "provided, test",
+  "com.google.guava" % "guava" % "22.0" % "provided",
```

Comment:
No problems, the version number won't be anyway leaking out.
The effect between old and new dependency is that do we want to verify that havarunner works with an old version (and trust that newer guava retains backwards compatibility) or do we test that it works with the latest released version.
I can revert that back if you want.
I do know that the ClassPath code is broken in guava 19, 20 and 21 versions - it fails when run inside jenkins workspaces that contain (by default) by default special characters (for example `workspace@2`)